docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,322 @@
+    "org.opencontainers.image.created": "2023-12-15T08:59:21Z"
+  }
+}
+```
+
+### oras discover
+
+View an artifact's referrers. The default output should be listed in a tree view.
+
+```bash
+oras discover $REGISTRY/$REPO:$TAG --format tree
+```
+
+```console
+localhost:localhost:5000/hello@sha256:5cb894d0c94c56894e160ad2eeb19a123b4d2155374e7709f43f8c0c2f249fe2
+├── application/vnd.oci.empty.v1+json
+│   └── sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8
+└── application/vnd.cncf.notary.signature
+    └── sha256:476d43120d3799fa76d7706f741beca73f5ff4149c8b6db3bd516a73d4a82fc1
+```
+
+View an artifact's referrers manifest in pretty JSON output. 
+
+```bash
+oras discover localhost:5000/hello:v1 --format json
+```
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8",
+      "size": 964,
+      "annotations": {
+        "org.opencontainers.image.created": "2023-12-14T13:48:32Z"
+      },
+      "artifactType": "application/vnd.oci.empty.v1+json"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:476d43120d3799fa76d7706f741beca73f5ff4149c8b6db3bd516a73d4a82fc1",
+      "size": 728,
+      "annotations": {
+        "io.cncf.notary.x509chain.thumbprint#S256": "[\"792265ec6b22f0a87c7b3d980319d51a76a382de1b7a47bd877bb4e5a9beb637\"]",
+        "org.opencontainers.image.created": "2023-12-14T14:41:56Z"
+      },
+      "artifactType": "application/vnd.cncf.notary.signature"
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> The `--format` flag will replace the existing `--output` flag. The `--output` will be marked as deprecated in ORAS v1.2.0. 
+
+## FAQ
+
+- Why choose to use `--format` flag to enable JSON formatted output instead of extending the existing `--output` flag?
+
+ORAS follows the [GNU](https://www.gnu.org/prep/standards/html_node/Option-Table.html#Option-Table) design principles. ORAS uses `--output` to output a file or directory and uses `--format` to format the output into JSON or using the given Go template. Popular tools like Docker, Podman, and Skopeo also follow this kind of design principles in the similar formatted output feature. 
+
+In addition, it will be a breaking change if we extend the `--output` flag to enable JSON formatted output. 
+
+- Why ORAS chooses [Go template](https://pkg.go.dev/text/template)?
+
+Go template is a powerful method to customize output you want It allows users to manipulate the output format of certain commands. It provides access to data objects and additional functions that are passed into the template engine programmatically. It also has some useful libraries that have strong functions for Go’s template language to manipulate the output data, such as [Sprig](https://masterminds.github.io/sprig/).
```

Comment:
Consider the following format
```suggestion
**Q:** Why choose to use `--format` flag to enable JSON formatted output instead of extending the existing `--output` flag?
**A:** ORAS follows the [GNU](https://www.gnu.org/prep/standards/html_node/Option-Table.html#Option-Table) design principles. ORAS uses `--output` to output a file or directory and uses `--format` to format the output into JSON or using the given Go template. Popular tools like Docker, Podman, and Skopeo also follow this kind of design principles in the similar formatted output feature.  In addition, it will be a breaking change if we extend the `--output` flag to enable JSON formatted output. 

**Q:** Why ORAS chooses [Go template](https://pkg.go.dev/text/template)?
**A:** Go template is a powerful method to customize output you want It allows users to manipulate the output format of certain commands. It provides access to data objects and additional functions that are passed into the template engine programmatically. It also has some useful libraries that have strong functions for Go’s template language to manipulate the output data, such as [Sprig](https://masterminds.github.io/sprig/).
```

Reply:
It renders:

**Q:** Why choose to use `--format` flag to enable JSON formatted output instead of extending the existing `--output` flag?
**A:** ORAS follows the [GNU](https://www.gnu.org/prep/standards/html_node/Option-Table.html#Option-Table) design principles. ORAS uses `--output` to output a file or directory and uses `--format` to format the output into JSON or using the given Go template. Popular tools like Docker, Podman, and Skopeo also follow this kind of design principles in the similar formatted output feature.  In addition, it will be a breaking change if we extend the `--output` flag to enable JSON formatted output. 

**Q:** Why ORAS chooses [Go template](https://pkg.go.dev/text/template)?
**A:** Go template is a powerful method to customize output you want It allows users to manipulate the output format of certain commands. It provides access to data objects and additional functions that are passed into the template engine programmatically. It also has some useful libraries that have strong functions for Go’s template language to manipulate the output data, such as [Sprig](https://masterminds.github.io/sprig/).